src/tasks/CloudFormationExecuteChangeSet/TaskOperations.ts

```diff
@@ -20,29 +20,50 @@ export class TaskOperations {
     ) {}
 
     public async execute(): Promise<void> {
-        const stackId = await this.verifyResourcesExist(
+        const changeSet = await this.verifyResourcesExist(
             this.taskParameters.changeSetName,
             this.taskParameters.stackName
         )
         let waitForUpdate = false
+        const stackId = changeSet.StackId || ''
         if (stackId) {
             waitForUpdate = await testStackHasResources(this.cloudFormationClient, this.taskParameters.stackName)
         }
 
-        console.log(tl.loc('ExecutingChangeSet', this.taskParameters.changeSetName, this.taskParameters.stackName))
-
         try {
-            await this.cloudFormationClient
-                .executeChangeSet({
-                    ChangeSetName: this.taskParameters.changeSetName,
-                    StackName: this.taskParameters.stackName
-                })
-                .promise()
+            const changeSetHasNoChanges =
+                changeSet.Status === 'FAILED' &&
+                changeSet.StatusReason ===
+                    "The submitted information didn't contain changes. Submit different information to create a change set."
```

Comment:
See thoughts in the main comment but at the very least we should tie this message (and the message in the test) to match https://github.com/aws/aws-toolkit-azure-devops/blob/62d2a6a786db05c5ea4ccae1e205713a8e89c1bc/src/tasks/CloudFormationCreateOrUpdateStack/TaskOperations.ts#L532 ; the CLI is also testing the same strings: https://github.com/aws/aws-cli/blob/45b0063b2d0b245b17a57fd9eebd9fcc87c4426a/awscli/customizations/cloudformation/deployer.py#L171-L174

Reply:
I've consolidated these checks into a utility function.